include/zim/archive.h

```diff
@@ -497,14 +497,13 @@ namespace zim
     public:
       explicit iterator(const std::shared_ptr<FileImpl> file, entry_index_type idx)
         : m_file(file),
-          m_idx(idx),
-          m_entry(nullptr)
+          m_idx(idx)
       {}
 
       iterator(const iterator<order>& other)
         : m_file(other.m_file),
           m_idx(other.m_idx),
-          m_entry(new Entry(*other.m_entry))
+          m_entry(new Entry(*other))
       {}
```

Comment:
Can you explain this change please ?

Reply:
The `m_entry` is only updated after we call `operator*()` on the iterator. So there is a potential risk of dereferencing a nullptr in `*other.m_entry` if we try to access it before calling the dereference operator. 

So to mitigate this risk, I think we should directly dereference `other` which will give us the entry.

Reply:
Ho yes. Good catch. Can you create a specific commit for this and put your explanation in the commit message please.

Reply:
Sure! thanks :D